tests/test_claude_wrapper_hooks.py

```
@@ -21,6 +22,22 @@
 
 
+def read_hook_pin() -> str:
+    """The pin cmux marks its own hook commands with, read from the wrapper.
+
+    Kept out of the tests as a literal so the two cannot drift: a wrapper that
+    stops pinning, or pins with a different token, fails these tests instead of
+    silently disabling hook ownership.
+    """
+    match = re.search(r"^cmux_claude_hook_pin='([^']+)'", SOURCE_WRAPPER.read_text(encoding="utf-8"), re.M)
+    if match is None:
+        raise AssertionError("cmux_claude_hook_pin is not defined in the wrapper")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)